src/Native/LibMFWrapper/mf.cpp

```diff
@@ -0,0 +1,1314 @@
+/*
```

Comment:
Can this be checked into our repo? 

Im assuming that we can't change this code then?

Reply:
Those files are included in TLC, so I think they can be checked in. Yes, we shouldn't modify those files from here (but we submit changes to the external group who is maintaining that package).

Reply:
is it possible to consume this dependency as a nuget? Alternatively, consume it as a subgit? Given that libmf and ML.NET are both OSS, that seems to be a cleaner solution than to copy the code.

Reply:
I will do submodule. Thanks.

---
In reply to: [226370835](https://github.com/dotnet/machinelearning/pull/1263#discussion_r226370835) [](ancestors = 226370835)

Reply:
As described in my reply to Markus below, LIBMF now is a submodule so we don't even need to check it in. We can change the code but it'd be better to submit PRs to that external repo.

---
In reply to: [225363641](https://github.com/dotnet/machinelearning/pull/1263#discussion_r225363641) [](ancestors = 225363641)